src/format/KdbxReader.cpp

```diff
@@ -95,6 +96,30 @@ bool KdbxReader::readDatabase(QIODevice* device, QSharedPointer<const CompositeK
         return false;
     }
 
+    auto authenticationFactorInfo = m_db->authenticationFactorInfo();
```

Comment:
This is one of the two changes to the existing database reading code: if we read information about authentication factors from the header, either add a MultiAuthenticationHeaderKey to the CompositeKey or replace the CompositeKey with one entirely.